src/Veldrid/Vk/VkDeviceMemoryManager.cs

```diff
@@ -67,8 +67,30 @@ internal unsafe class VkDeviceMemoryManager : IDisposable
             VkImage dedicatedImage,
             Vulkan.VkBuffer dedicatedBuffer)
         {
-            // Round up to the nearest multiple of bufferImageGranularity.
-            size = ((size / _bufferImageGranularity) + 1) * _bufferImageGranularity;
+            if (dedicated)

```

Comment:
This was quarter-baked and I guarantee it's insufficient.  If anyone has suggestions on how this should be implemented, let me know.  Otherwise I'll have to dive in and try and figure it out from looking closer at the docs.

Reply:
Looking at the spec, it seems that sparse resource sizes must be a multiple of buffer-image granularity.
I am quite certain that dedicated resource sizes do not need to be a multiple of that granularity.

[In my fork](https://github.com/TechnologicalPizza/veldrid/blob/c65946a7e1caeece088e2433b5676940dd02e25b/src/Veldrid/Vk/VkDeviceMemoryManager.cs#L84) I've applied this as per spec and it works on Nvidia and AMD hardware.

I would also like to point out that the current alignment equation is flawed and will waste by one multiple when the value is `x + 1 = multiple` (look in my linked fork for a corrected equation).